lib/internal/readline/eventsToAsyncIteratorFactory.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+const {
+  Promise,
+  SymbolAsyncIterator,
+} = primordials;
+
+const PAUSE_THRESHOLD = 1024;
+const RESUME_THRESHOLD = 1;
+const ITEM_EVENTS = ['data'];
+const CLOSE_EVENTS = ['close', 'end'];
+const ERROR_EVENTS = ['error'];
+
+let FixedQueue;
+let setImmediate;
+function waitNext(emitter, next, events) {
+  return new Promise((resolve, reject) => {
+    const resolveNext = () => {
+      events.forEach((event) => emitter.off(event, resolveNext));
+      setImmediate(() => {
+        try {
+          resolve(next());
+        } catch (promiseError) {
+          reject(promiseError);
+        }
+      });
+    };
+    events.forEach((event) => emitter.once(event, resolveNext));
+  });
+}
+
+function turnFactory(readable, callback, onOff) {
+  return (event) => readable[onOff](event, callback);
+}
+
+module.exports = function eventsToAsyncIteratorFactory(readable, {
+  pauseThreshold = PAUSE_THRESHOLD,
+  resumeThreshold = RESUME_THRESHOLD,
+  closeEvents = CLOSE_EVENTS,
+  itemEvents = ITEM_EVENTS,
+  errorEvents = ERROR_EVENTS
+}) {
+  const events = [...itemEvents, ...errorEvents, ...closeEvents];
```

Comment:
Avoid spread operator on arrays, which relies on globally mutable `Array.prototype[Symbol.iterator]`. Use `primordials.ArrayPrototypeConcat` instead.
```suggestion
  const events = ArrayPrototypeConcat(itemEvents, errorEvents, closeEvents);
```